src/Helper/ArchiveHelper.php

```diff
@@ -322,15 +323,24 @@ private function createSubCase(string $caseId, string $caseName): string {
    */
   private function uploadDocumentToCase(string $caseId, string $webformAttachmentElementId, WebformSubmission $submission, bool $shouldArchiveFiles, bool $shouldBeFinalized): void {
     // Handle main document (the attachment).
-    $element = $submission->getWebform()->getElement($webformAttachmentElementId);
-    $fileContent = WebformEntityPrintAttachment::getFileContent($element, $submission);
+    $webformAttachmentElement = $submission->getWebform()->getElement($webformAttachmentElementId);
+    $fileContent = WebformEntityPrintAttachment::getFileContent($webformAttachmentElement, $submission);
+    $webformLabel = $submission->getWebform()->label();
+    $pdfExtension = '.pdf';
 
-    // Ids that should possibly be finalized (jornaliseret) later.
-    $documentIdsForFinalizing = [];
+    if ($webformAttachmentElement['#filename']) {
+      if ($webformAttachmentElement['#type'] === 'webform_entity_print_attachment:pdf') {
+        $baseName = WebformAttachmentBase::getFileName($webformAttachmentElement, $submission);
+        $getOrganizedFileName = substr_replace($baseName, '-' . $webformLabel . '-' . $submission->serial(), strrpos($baseName, $pdfExtension), 0);
```

Comment:
A comment telling what's going on here would be much appreciated.

Reply:
See other response.